src/agent/src/config.rs

```diff
@@ -239,16 +239,22 @@ impl AgentConfig {
         let mut config: AgentConfig = Default::default();
         let cmdline = fs::read_to_string(file)?;
         let params: Vec<&str> = cmdline.split_ascii_whitespace().collect();
+
+        // Check if there is config file before parsing params that might
+        // override values from the config file.
         for param in params.iter() {
             // If we get a configuration file path from the command line, we
             // generate our config from it.
             // The agent will fail to start if the configuration file is not present,
             // or if it can't be parsed properly.
             if param.starts_with(format!("{}=", CONFIG_FILE).as_str()) {
                 let config_file = get_string_value(param)?;
-                return AgentConfig::from_config_file(&config_file);
+                config = AgentConfig::from_config_file(&config_file)?;
+                break;
```

Comment:
I think I understand what's happening here and why we're not being able to block the `ExecProcessRequest`.

Before this patch, we'd get the agent configuration and simply return, not going through the rest of this function.  However, with this patch what happens is that we end up reaching the following line at the end of this function: https://github.com/kata-containers/kata-containers/blob/26075bc9793144341894dc5bdeed40216c974a5a/src/agent/src/config.rs#L351-L352

The simplest way to solve this would adding a way to check whether the default agent config was used, and then only set the `config.endpoints.all_allowed = true;` in case we're did **not** get an agent configuration file as a parameter.

@fitzthum, mind to rework this bit and ensure we cover this as part of our tests?

Reply:
I wonder if an even simpler fix might be to just check `if config.endpoints.allowed.is_empty()` around L352. The only problem I can see there is if we've said in the agent config that we don't want any endpoints to be allowed, but I'm pretty sure the agent wouldn't function then

Reply:
@fitzthum - not that I'm in a made rush to get downstream changes in after this or anything before a deadline, but I've created a PR with a unit test and possible fix: https://github.com/fitzthum/kata-containers/pull/1 @fidencio - your input would be welcome too.